platforms/ide/tooling-api/src/main/java/org/gradle/tooling/model/dsl/KotlinDslBaseScriptModel.java

```diff
@@ -53,4 +53,12 @@ public interface KotlinDslBaseScriptModel {
     @Incubating
     List<String> getImplicitImports();
 
+    /**
+     * Template classes used for interpreting Kotlin DSL scripts.
+     *
+     * @since 9.2.0
+     */
+    @Incubating
+    List<String> getTemplateClassNames();
```

Comment:
❓ What would the TAPI clients (IDEs) use this for?

With the implementation in this PR it contains a bag of classes for legacy templates, new templates and some resolver that is also legacy. It feels like it pushes a test utility to a production model.

I would simply remove that from this PR.

Reply:
Got it. Thank you for the context.
Still, the list currently contains useless classes. I think what JetBrains is after is only the templates (not the resolver). Moreover I think we should just return the set of new templates, not the legacy ones.

Reply:
Made the change to only return the current templates (so removed the legacy ones and the resolver).